api/src/paths/collections/index.ts

```
@@ -0,0 +1,330 @@
+
+const defaultLog = getLogger('paths/collections');
+
+const collectionService = new CollectionService();
+
+/**
+ * GET /api/collection
+ * Fetch all collections.
+ */
+export const GET: Operation = async (req: Request, res: Response) => {
+  const connection = getAPIUserDBConnection();
+
+  try {
+    await connection.open();
+
+    const collections = await collectionService.getAllCollections(connection);
+
+    await connection.commit();
+
+    res.status(200).json(collections);
+  } catch (error) {
+    defaultLog.error({ label: 'GET /api/collection', message: 'error', error });
+    connection.rollback();
+    res.status(500).json({ error: 'Failed to fetch collections' });
+  } finally {
+    connection.release();
+  }
+};
+
+GET.apiDoc = {
+  description: 'Fetch all collections.',
+  tags: ['collection'],
+  responses: {
+    200: {
+      description: 'List of collections.',
+      content: {
+        'application/json': {
+          schema: {
+            type: 'array',
+            items: {
+              type: 'object',
+              properties: {
+                collection_id: { type: 'integer' },
+                name: { type: 'string' },
+                objectives: { type: 'string' }
+              }
+            }
+          }
+        }
+      }
+    },
+    500: {
+      description: 'Internal server error.'
+    }
+  }
+};
+
+/**
+ * POST /api/collection
+ * Create a new collection.
+ */
+export const POST: Operation = async (req: Request, res: Response) => {
+  const connection = getAPIUserDBConnection();
+
+  try {
+    await connection.open();
+
+    const newCollection = await collectionService.createCollection(connection, req.body);
+
+    await connection.commit();
+
+    res.status(201).json(newCollection);
+  } catch (error) {
+    defaultLog.error({ label: 'POST /api/collection', message: 'error', error });
+    connection.rollback();
+    res.status(500).json({ error: 'Failed to create collection' });
```

Throw an error instead of returning a 500